File: native/V2StyxLib/src/messages/base/StyxTMessageFID.cpp
/*
 * StyxTMessageFID.cpp
 *
 *  Created on: Apr 07, 2018
 *      Author: vschryabets@gmail.com
 */

#include "messages/base/StyxTMessageFID.h"
#include <sstream>

StyxTMessageFID::StyxTMessageFID(MessageTypeEnum type, MessageTypeEnum answer, StyxFID fid)
	: StyxTMessage(type, answer), mFID(fid) {
}

StyxTMessageFID::~StyxTMessageFID() {
}

StyxFID StyxTMessageFID::getFID() {
	return mFID;
}

void StyxTMessageFID::setFID(StyxFID fid) {
	mFID = fid;
}

void StyxTMessageFID::load(IStyxDataReader *buffer) {
	StyxTMessage::load(buffer);
	mFID = buffer->readUInt32();
}

void StyxTMessageFID::writeToBuffer(IStyxDataWriter *outputBuffer) {
	StyxTMessage::writeToBuffer(outputBuffer);
	outputBuffer->writeUInt32(mFID);
}

size_t StyxTMessageFID::getBinarySize() {
	return StyxTMessage::getBinarySize() + sizeof(StyxFID);
}

StyxString StyxTMessageFID::toString() {
	StyxString result = StyxTMessage::toString();
	std::stringstream stream;
	stream << result << "\tFID: " << mFID;
	return stream.str();
}

File: java/library/src/main/java/com/v2soft/styxlib/messages/base/StyxTMessageFID.java
package com.v2soft.styxlib.messages.base;

import com.v2soft.styxlib.io.IStyxDataReader;
import com.v2soft.styxlib.io.IStyxDataWriter;
import com.v2soft.styxlib.messages.base.enums.MessageType;

import java.io.IOException;
import java.io.UnsupportedEncodingException;

public class StyxTMessageFID extends StyxTMessage {
    private long mFID;
    // ===========================================================================
    // Constructor
    // ===========================================================================
    public StyxTMessageFID(MessageType type, MessageType answer, long fid) {
        super(type, answer);
        mFID = fid;
    }
    // ===========================================================================
    // Getters
    // ===========================================================================
    public long getFID() {
        return mFID;
    }
    // ===========================================================================
    // Setters
    // ===========================================================================
    public void setFID(long fid) {
        mFID = fid;
    }
    // ===========================================================================
    // Encoder/Decoder methods
    // ===========================================================================
    @Override
    public void load(IStyxDataReader input)
            throws IOException  {
        super.load(input);
        mFID = input.readUInt32();
    }

    @Override
    public int getBinarySize() {
        return super.getBinarySize() + 4;
    }

    @Override
    public void writeToBuffer(IStyxDataWriter output)
            throws UnsupportedEncodingException, IOException {
        super.writeToBuffer(output);
        output.writeUInt32(mFID);
    }

    @Override
    public String toString() {
        String result = super.toString();
        return String.format("%s\tFID: %d", result, getFID());
    }
}
